azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,488 @@
+        if object_text:
+            objects_obj = yaml.safe_load(object_text)
+        else:
+            objects_obj = {"array": []}
+        entry_text = yaml.safe_dump(secret_entry, indent=6)
+        objects_obj["array"].append(entry_text)
+        object_text = yaml.safe_dump(objects_obj, indent=6)
+        # TODO: formatting will be removed once fortos service fixes the formatting issue
+        return object_text.replace("\n- |", "\n    - |")
+
+    def _get_cl_resources(self, instance_name: str, resource_group: str) -> dict:
+        self.instance = self.instances.show(name=instance_name, resource_group_name=resource_group)
+        self.resource_map = self.instances.get_resource_map(self.instance)
+        custom_location = self.resource_client.resources.get_by_id(
+            resource_id=self.instance["extendedLocation"]["name"], api_version=CUSTOM_LOCATIONS_API_VERSION
+        )
+        cl_resources = self.resource_map.connected_cluster.get_aio_resources(custom_location_id=custom_location["id"])
+        return cl_resources
+
+    def _find_existing_spc(self, instance_name: str, cl_resources: List[dict]) -> dict:
+        # check if secret sync enabled by getting the default secretproviderclass
+        secretsync_spc = None
+
+        if cl_resources:
+            for resource in cl_resources:
+                if resource["type"].lower() == "microsoft.secretsynccontroller/azurekeyvaultsecretproviderclasses":
+                    resource_id_container = parse_resource_id(resource["id"])
+                    secretsync_spc = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
+                        resource_group_name=resource_id_container.resource_group_name,
+                        azure_key_vault_secret_provider_class_name=resource_id_container.resource_name,
+                    )
+                    break
+
+        if not secretsync_spc:
+            raise ResourceNotFoundError(
+                f"Secret sync is not enabled for the instance {instance_name}. "
+                "Please enable secret sync before adding certificate."
+            )
+
+        return secretsync_spc
+
+    def _check_and_update_secret_name(self, secrets: PageIterator, secret_name: str, spc_keyvault_name: str) -> str:
+        from rich.prompt import Confirm, Prompt
+
+        new_secret_name = secret_name
+        for secret in secrets:
+            if secret.id.endswith(secret_name):
+                # Prompt user to decide on overwriting the secret
+                overwrite_secret = Confirm.ask(
+                    f"Secret with name {secret_name} already exists in keyvault {spc_keyvault_name}. "
+                    "Do you want to overwrite the secret name?",
+                )
+
+                if overwrite_secret:
+                    new_secret_name = Prompt.ask("Please enter the new secret name")
+
+                return new_secret_name
```

Comment:
Is there any way for users to `--force` or `--yes` to avoid these prompts? If this occurs in any sort of automation scenario these may block

Reply:
Discussed offline and we can address this in a follow-up PR before releasing. Let's get the code in and work through the logic for this afterwards